helix-view/src/view.rs

```diff
@@ -102,6 +102,38 @@ pub struct ViewPosition {
     pub vertical_offset: usize,
 }
 
+#[derive(Clone)]
+pub struct SearchMatches {
+    pub matches: Vec<(usize, usize)>,
+    pub new: Regex,
+    pub old: Regex,
+}
+
+impl fmt::Debug for SearchMatches {
+    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+        f.debug_struct("SearchMatches")
+            .field("matches", &self.matches)
+            .field("new", &self.new)
+            .field("old", &self.old)
+            .finish()
+    }
+}
+
+impl SearchMatches {
+    pub fn clear(&mut self) {
+        self.matches.clear();
+    }
+
+    pub fn has_changed(&self, regex: &Regex) -> bool {
+        regex.as_str() != self.old.as_str()
+    }
```

Comment:
could not figure out a clean way to compare `Regex`, so this is what i came up with